### PR TITLE
Update merkato.py

### PR DESCRIPTION
**Pull request by needmoney90**

### CODE REVIEW

**Comment by needmoney90** [comment]
This allows users to set a profit-take before order replacement. Believe this does it @15chrjef

**Comment by needmoney90** [comment]
I'll let you approve before merging.

File: merkato/merkato.py [hunk 1]
```diff
@@ -74,14 +78,15 @@ def rebalance_orders(self, new_txes):
             if tx['type'] == SELL:
                 if DEBUG: print(SELL) #todo: change # to new format
                 # print('amount', type(tx['amount']), type(tx[PRICE])) # todo use debug
-                amount = float(tx['amount']) * float(tx[PRICE])
+                
+                amount = float(tx['amount']) * float(tx[PRICE])*(1-factor)
```
Comment (nasaWelder) [inline merkato/merkato.py]: Isn't amount always denominated in coin units?

**Review by mark-antony1 — COMMENTED** [review]
So this looks like it factor removes an amount of the last sale but not necessarily the profit. 

Wouldn't the profit would be half the spread, so the factor should take some portion of that chunk?

Also there is no implementation of how the profit margin would be used, I don't know how you were thinking the rest of merkato would change to use it, but i'm assuming there would need to be a new field in the merkatos table, and a function to update it, and a script that runs that function.

**Comment by nasaWelder** [comment]
Also, profit must be known exactly, and then a factor applied to that amount, otherwise you could non-sustainably pull out money, until you had none to trade with. That's not fatal, you just take yourself out of the liquidity business.

In order to know exactly what profit is, a trade would need to know if it's first, or if it's got an "ancestor". This isn't impossible, but I think it's a major architecture development, maybe we need a new sub object, or at least a robust dictionary. You wouldn't want to keep all that in memory for all time, just the previous ancestor, I think.

**Comment by needmoney90** [comment]
Nasa, the profit is half the spread

**Comment by mark-antony1** [comment]
Its implemented properly in the rebalance_orders now it needs to be included in the gui, merkato init, added to the merkatos table, and anything else we need.

**Comment by needmoney90** [comment]
Fixed it to have half the spread instead of the whole thing

**Comment by needmoney90** [comment]
made some changes and pushed by accident, the diff history looks broken but is actually fine...

File: merkato/merkato.py (hunk 1 above)
Comment (needmoney90) [reply merkato/merkato.py]: yes, so in this one we need to convert to the format needed for the buy

**Pull request merged**